weblate/screenshots/models.py

```diff
@@ -76,3 +90,51 @@ def change_screenshot_assignment(sender, instance, action, **kwargs):
         name="UnusedScreenshot"
     ).exists():
         component_alerts.delay([instance.pk])
+
+
+@receiver(vcs_post_update)
+def update_screenshots_from_repo(sender, component, previous_head: str, **kwargs):
+    repository = component.repository
+    last_remote_revision = repository.last_remote_revision
+    changed_files = repository.get_changed_files(compare_to=previous_head)
+    screenshots = Screenshot.objects.filter(
+        translation__component=component, repository_filename__in=changed_files
+    )
+
+    # Update existing screenshots
+    for screenshot in screenshots:
+        filename = screenshot.repository_filename
+        changed_files.remove(filename)
+
+        file_data = repository.get_file(filename, last_remote_revision, raw=True)
+        image_file = File(file_data, name=filename)
```

Comment:
```suggestion
        full_name = os.path.join(component.full_path, filename)
        image_file = File(open(full_name, "rb"))
```

There is no need to go through the repository as we want current content.

Reply:
okay, so we just want the link to file, not actually downloading it and saving it in the VCS?

Reply:
The file is already present in the Git checkout, so you can open it directly, instead of using `git show` to obtain the content.

Reply:
I tested with getting the file from Git checkout and I see this error for the path traversal. I think the path `component.full_path` is absolute and not relative, that maybe the reason?
 
<img width="1645" alt="Screenshot 2023-04-05 at 11 48 45" src="https://user-images.githubusercontent.com/24358501/230045658-42c5f184-2e06-4f2f-8e11-3ab7787317d3.png">

Reply:
You probably should pass `name=os.path.basename(filename)` to `File` anyway to avoid including possibly long paths from the repository there.